docs/architecture/adr-004-genesis-modes.md

```diff
@@ -0,0 +1,253 @@
+# ADR 4: Multiple Genesis modes
+
+## Changelog
+
+- Sept 3, 2019: Initial draft
+
+## Context
+
+In the current SDK, when the `GenesisState` is initialized through a `InitGenesis` call, some of the `GenesisState` fields are loaded from a `genesis.json` file, whereas in other cases they are initialized (_i.e_ populated) if the corresponding object value from JSON is empty, undefined or its value is zero.
+
+This may lead to confusion because as is currently programmed, `InitGenesis`
```

Comment:
I still don't understand the problem or context here? I took a look at every single `InitGenesis` method and all it does is takes input genesis state and sets the corresponding values/params on the respective keeper.

Reply:
Yeah I'm not sure this paragraph makes total sense. I think what is intended to be said is that existing genesis will populate existing fields even if they were accidentally left out, which can be problematic